torch/_lobpcg.py

```diff
@@ -536,7 +536,8 @@ def _get_rayleigh_ritz_transform(self, S):
         SBS = _utils.qform(B, S)
 
         d1 = SBS.diagonal(0, -2, -1) ** -0.5
-        d = d1.reshape(d1.shape[0], 1)
+        d = torch.zeros((d1.shape[0], 1), dtype=d1.dtype, device=d1.device)
```

Comment:
it's quite strange that the previous version memleaked... if it's the case, maybe it's worth to file a bug about it?

Reply:
I have not been able to create a simple reproducer: while it seems to be related to using `reshape`, it may also be the case where the bug only shows up when using the `reshape`.
However, the cuda memory leak is reproducible by disabling the `if`-block in https://github.com/Quansight/pytorch/blob/81524a7b2757a8877637cd0c5e3cb829207b5c3f/torch/_lobpcg.py#L551 .

I suggest filing a bug report after merging this PR in order to make the bug easier to reproduce.